src/core_modules/capture-core/components/Pages/Enrollment/EnrollmentPageDefault/runRulesForEnrollment.js

```diff
@@ -0,0 +1,109 @@
+// @flow
+import { convertValue } from 'capture-core/converters/serverToClient';
+import runRulesForEnrollmentPage from 'capture-core/rules/actionsCreator/runRulesForEnrollmentPage';
+import { dataElementTypes } from '../../../../metaData';
+import type { InputRuleEnrollmentData } from './types/common.types';
+
+// $FlowFixMe
+const convertDate = (date: string): string => convertValue(date, dataElementTypes.DATE);
+
+const getDataElementsFromProgram = (data, eventsDataValues) => {
+    if (!data?.programStages) { return {}; }
+    return data.programStages
+        .reduce((acc, stage) => {
+            stage.programStageDataElements.forEach((programStageDataElement) => {
+                const { id, valueType, optionSet } = programStageDataElement.dataElement;
+                const { value } = eventsDataValues.find(item => item.dataElement === id) || { };
+                acc[id] = {
+                    id,
+                    valueType,
+                };
+                if (optionSet) {
+                    acc[id].optionSetId = optionSet.id;
+                }
+                if (value) {
+                    acc[id].value = convertValue(value, valueType);
+                }
+            });
+            return acc;
+        }, {});
+};
+
+const getEventValuesFromEvent = (enrollment, eventId, dataElements) => {
+    const currentEvent = enrollment.events.find(item => item.event === eventId);
+    const eventValues = {};
+    if (currentEvent) {
+        currentEvent.dataValues.forEach((dataValue) => {
+            const { dataElement } = dataValue;
+            eventValues[dataElement] = dataElements[dataElement].value;
+        });
+    }
+
+    return eventValues;
+};
+
+
+const getEventsDataFromEnrollment = (enrollment, dataElements) => {
+    const allEvents = enrollment.events.map(event => ({
+        eventId: event.event,
+        programId: event.program,
+        programStageId: event.programStage,
+        orgUnitId: event.orgUnit,
+        orgUnitName: event.orgUnitName,
+        trackedEntityInstanceId: event.trackedEntityInstance,
+        enrollmentId: event.enrollment,
+        enrollmentStatus: event.enrollmentStatus,
+        status: event.status,
+        eventDate: convertDate(event.dueDate),
+        dueDate: convertDate(event.dueDate),
+        ...getEventValuesFromEvent(enrollment, event.event, dataElements),
+    }));
+
+    const groupByProgramStageId = allEvents.reduce((acc, currentEvent) => {
+        acc[currentEvent.programStageId] = { ...currentEvent };
```

Comment:
we need an array of events here (multiple events can have the same programStageId)